src/main/java/cn/nukkit/registry/BlockRegistry.java

```diff
@@ -508,6 +508,12 @@ private void registerVanillaBlocks() {
 
         this.factoryMap.put(CAMPFIRE, BlockCampfire::new);
 
+        this.factoryMap.put(ACACIA_TRAPDOOR, BlockTrapdoorAcacia::new); //400
+        this.factoryMap.put(BIRCH_TRAPDOOR, BlockTrapdoorBirch::new); //401
+        this.factoryMap.put(DARK_OAK_TRAPDOOR, BlockTrapdooDarkOak::new); //402
+        this.factoryMap.put(JUNGLE_TRAPDOOR, BlockTrapdoorJungle::new); //403
+        this.factoryMap.put(SPRUCE_TRAPDOOR, BlockTrapdoorSpruce::new); //404
+
```

Comment:
On discord, @Braayy suggested that in `2.0` we don't need to create new blocks classes in this case, that the PR could be reduced to:
```java
        this.factoryMap.put(ACACIA_TRAPDOOR, BlockTrapdoor::new); //400
        this.factoryMap.put(BIRCH_TRAPDOOR, BlockTrapdoor::new); //401
        this.factoryMap.put(DARK_OAK_TRAPDOOR, BlockTrapdoor::new); //402
        this.factoryMap.put(JUNGLE_TRAPDOOR, BlockTrapdoor::new); //403
        this.factoryMap.put(SPRUCE_TRAPDOOR, BlockTrapdoor::new); //404
```
Like what happens with the different door types.

But since the files are already created and we have other examples that creates new classes, like the stairs blocks, can you add more value to the new classes by mapping the color they have?

You can use `BlockWood` and `BlockWood2` from the `master` branch as reference for the block colors. The `BlockLog` and `BlockLog2` from `2.0` is missing some colors.

Reply:
Fixed at 27e9728ea6b3d1c6c4c0be295ccf9ad379ec5153